dts/bindings/misc/nxp,imx-flexio.yaml

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2023, STRIM, ALC
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP i.MX FlexIO controller
+
+compatible: "nxp,imx-flexio"
+
+include: base.yaml
```

Comment:
Could you create yaml for define device classification?
raspi_pico has configurable-IO function that is similar to flexio.
#44316

I think it is enough with just place holder for define classification.
No need to define API because these devices used from their client drivers.

@danieldegrasse @teburd @yonsch 
Please comment if you have any opinion.
I would like to consider implementing rpi_pico to reflect this.

Reply:
@soburi Not sure I understand this correctly, do you mean changing the current PIO PR? Because currently I don't see how it's different from what you're suggesting. Anyway, it should be discussed in the PIO PR.
If you meant changing the flex IO PR to be more like the PIO, then I generally agree, though I'm not really familiar with the flex IO enough to say. If possible, I do think that the generic driver should not require modifying it for every child device, but rather be something that child drivers can inherit from/incorporate into.

Reply:
@yonsch 
Sorry for the confusion.

I thought that if a common definition was created, it may affect PIO, so I notified it.

(If it is to be standardized, I also think it would be better to join the discuss how to make it a "preferred" form based on the knowledge of PIO.)

Reply:
@soburi this will be discussed in next week's architecture WG meeting

Reply:
@soburi I think these devices probably do belong in "misc". I am less familiar with the PIO, but it is programmed with a set of instructions, correct? The FlexIO may be able to implement similar Zephyr driver classes, but it is not configured in the same manner. I think misc drivers make sense here, because the API to manage resources for each peripheral will be very specific to that peripheral (and may not always be required), so attempting to find similarities in the APIs to manage these peripherals does not make much sense to me. I think the primary similarity between these peripherals is that they can emulate some of the same driver classes in Zephyr.